### PR TITLE
Fix different return type in .c and .h header file (BSP-250)

### PR DESCRIPTION
**Pull request by hayschan**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [x] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [ ] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [ ] New files were added to CI build job
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Change description

Fix return type declaration in header file but should have returned esp_err_t instead.

Return type `void` is declared in header file.

https://github.com/espressif/esp-bsp/blob/f9a521aaf96722e4ea0bf43794eaebf80388f5e4/SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h#L71

https://github.com/espressif/esp-bsp/blob/f9a521aaf96722e4ea0bf43794eaebf80388f5e4/SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h#L77

However, in `ws_7inch.c`, `esp_err_t` is written as the return type for `bsp_i2c_init()` and `bsp_i2c_deinit()`.

https://github.com/espressif/esp-bsp/blob/f9a521aaf96722e4ea0bf43794eaebf80388f5e4/SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/ws_7inch.c#L62

https://github.com/espressif/esp-bsp/blob/f9a521aaf96722e4ea0bf43794eaebf80388f5e4/SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/ws_7inch.c#L78

### CODE REVIEW

File: SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h [hunk 1]
```diff
@@ -68,13 +68,13 @@ extern "C" {
  * @brief Init I2C driver
  *
  */
-void bsp_i2c_init(void);
+esp_err_t bsp_i2c_init(void);
```
Comment (espzav) [inline SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h:71]: Please, update the comment:
```
 *
 * @return
 *      - ESP_OK                On success
 *      - ESP_ERR_INVALID_ARG   I2C parameter error
 *      - ESP_FAIL              I2C driver installation error
```

File: SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h [hunk 2]
```diff
@@ -68,13 +68,13 @@ extern "C" {
...
 
 /**
  * @brief Deinit I2C driver and free its resources
  *
  */
-void bsp_i2c_deinit(void);
+esp_err_t bsp_i2c_deinit(void);
```
Comment (espzav) [inline SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h:77]: Please, update the comment:
```
 *
 * @return
 *      - ESP_OK                On success
 *      - ESP_ERR_INVALID_ARG   I2C parameter error
```

**Review by espzav — COMMENTED** [review]
@hayschan Thank you for the fix this issue!

File: SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h (hunk 2 above)
Comment (hayschan) [reply SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h:84]: Should there be ESP_FAIL as well?

**Comment by espzav** [comment]
@hayschan Please use pre-commit hook: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/contribute/install-pre-commit-hook.html

File: SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h (hunk 2 above)
Comment (espzav) [reply SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/include/bsp/ws_7inch.h:84]: There should be all error codes, which can be returned. Then yes please.

**Review by tore-espressif — APPROVED** [review]

**Comment by espzav** [comment]
@hayschan Any update?

**Comment by hayschan** [comment]
> @hayschan Please use pre-commit hook: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/contribute/install-pre-commit-hook.html

I have installed pre-commit hook following the guide.

I installed by running `pre-commit install --allow-missing-config` at my local "hayschan/esp-bsp" folder. 

What do I need to do next? I can't `git add` nor `git commit` as I make no changes after installing pre-commit

**Comment by espzav** [comment]
Hi @hayschan,
I made this change in this PR: https://github.com/espressif/esp-bsp/pull/126 we can close this PR. Thank you for your help!

**Comment by hayschan** [comment]
> Hi @hayschan, I made this change in this PR: #126 we can close this PR. Thank you for your help!

Thanks.

> @hayschan you can use tool for modified files: `tools/format.sh yourfile.c` more is here: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/contribute/style-guide.html#formatting-your-code

I don't have a Linux/Unix based computer on hand. Could not run the format.sh on Windows.

**Pull request closed without merging**